### ISSUE TITLE
Support reprogrammable keys

### ISSUE THREAD

**Issue by Lekensteyn** [pwr-Solaar/Solaar#31]
As requested in http://superuser.com/q/466253/47108.

The keyboards seem to allow changing the functionality of Fn + Fx keys, instead of "Email", another event is generated (I haven't checked whether this is a regular keymapping that is sent to the input layer or only a HID++ notification).

I will probably start working on this next week.

**Comment by pfps** [comment]
I have an initial implementation of reprogrammable keys for Solaar. Reprogramming a key or button tells a device that the key or button should act like another key or button (even a key or button that is not on the device at all).  Only keys and buttons that are reprogrammable (not divertable - that's something that needs support from drivers) can be reprogrammed, and they can only be reprogrammed to a limited set of targets.

The interface presents two selection boxes.  The left-hand box shows a key or button that can be reprogrammed.  The right-hand box has the possible targets for that key or button, with the default target first.  To change the behaviour of a key or button, select the key in the left-hand box and then select the target in the right-hand box.

To test out this feature, clone Solaar from my directory and run from its keys branch, via
```
git clone https://github.com/pfps/Solaar
cd Solaar
git checkout keys
bin/solaar
```

A few caveats:

- This is a preliminary implementation and still has debugging statements in it, but the implementation is working for me so it is time to have it tested out by others and on other hardware.

- The implementation is only for the feature REPROG CONTROLS V4 {1B04}. Running `solaar show` will show all the features of your device.  If you have a device with a different version of REPROG CONTROLS let me know and I may be able to implement that version as well.

- The implementation is slower than it should be because it communicates too much with the device.  I expect to fix this in a little while.

- The interface is not nearly as slick as the Logitech one, and it never will be.

- The key and button names and the target names are taken from Logitech documentation and may not be intuitive.  Suggestions for improvements are welcome.

- There is a preliminary implementation of REPORT RATE here as well, but changing the rate doesn't work.

**Comment by pfps** [comment]
Closed by PR #810
